### PR TITLE
Manual code review

### LINKED ISSUE #13: Manual code review

### PR DESCRIPTION
**Pull request by kriman**

### CODE REVIEW

File: src/components/scientist/Scientist.java [hunk 1]
```diff
@@ -90,15 +90,13 @@ public void move() {
      */
     public boolean learn(GeneticCode gc) {
         Skeleton.printCall("Scientist.learn()");
-        if (inventory.validateAction(LEARN) == LEARN) {
+        if (inventory.validateAction(LEARN) == LEARN && inventory.add(gc)) {
             // Scientist tud tanulni
             // hozzáadás inventoryhoz. az inventory eldobja a duplikátumot
-            if(inventory.add(gc)) {
                 // az inventorytól elkéri a megtanult genetikai kódokat
```
Comment (Aksurtep9) [inline src/components/scientist/Scientist.java]: Comments are not necesarry ;)

File: src/components/scientist/Scientist.java [hunk 2]
```diff
@@ -90,15 +90,13 @@ public void move() {
      */
     public boolean learn(GeneticCode gc) {
         Skeleton.printCall("Scientist.learn()");
-        if (inventory.validateAction(LEARN) == LEARN) {
```
Comment (Aksurtep9) [inline src/components/scientist/Scientist.java:93]: This if is mergable with the child if statement

File: src/components/scientist/Scientist.java [hunk 3]
```diff
@@ -90,15 +90,13 @@ public void move() {
...
+        if (inventory.validateAction(LEARN) == LEARN && inventory.add(gc)) {
             // Scientist tud tanulni
             // hozzáadás inventoryhoz. az inventory eldobja a duplikátumot
-            if(inventory.add(gc)) {
```
Comment (Aksurtep9) [inline src/components/scientist/Scientist.java:96]: This if is mergable with its parent if statement

**Review by Aksurtep9 — APPROVED** [review]
commint review finished

**Review by Aksurtep9 — APPROVED** [review]
commit review finished

**Review by Aksurtep9 — APPROVED** [review]
commit review finished

**Review by Aksurtep9 — APPROVED** [review]
commit review finished

**Review by Aksurtep9 — APPROVED** [review]
commit review finished

**Review by Aksurtep9 — APPROVED** [review]
commit review finished

**Comment by kriman** [comment]
@Aksurtep9 lgtm!

**Pull request merged**